lib/core/src/Cardano/Wallet.hs

```diff
@@ -720,6 +732,18 @@ newWalletLayer tracer bp db nw tl = do
             (w, _) <- _readWallet wid
             DB.putCheckpoint db (PrimaryKey wid) (newPending (tx, meta) w)
 
+    _slotIdTime
+        :: SlotId
+        -> UTCTime
+    _slotIdTime slN =
+        addUTCTime (convert $ slotLength * slots) block0Date
+      where
+        slots = fromIntegral . fromEnum $ (flatSlot slN) - (flatSlot sl0)
+        sl0 = block0 ^. #header ^. #slotId
+
+        convert :: DiffTime -> NominalDiffTime
+        convert = toEnum . fromEnum
```

Comment:
we can also use the fact that that `NominalDiffTime` and `DiffTime` both have `RealFrac` instances. Hence they have both `Real` and `Fractional` instances. If so we can ,
```
convert :: DiffTime -> NominalDiffTime
convert = fromRational . toRational
```